### ISSUE TITLE
Select2 doesn't work in embeded edit view (subform)

### ISSUE THREAD

**Issue by yoson7** [saltcorn/saltcorn#2935]
Using embedded edit view (sub form) that include select2 field.

Steps to reproduce:

Create Table A for groups (or cities or departments), and table B for related members (like group members, or streets in the city or department members) and table C for common names for members or streets.
Create edit view for table B, the edit view includes a select2 input field to select the name from table C.
Create edit view for table A and embed the edit view for table B - to create\edit the group and its members.
Press the + sign to have the forms of 3 members in the main form of the group.
Now you find that you can't edit the names of the members, only the name in the top form.

Why the + to add sub form is duplicating the existing filled form, instead of creating empty fresh sub form?

**Comment by christianhugoch** [comment]
Hi, this happens because we're cloning the full initialized select2 elements. I believe we should remove all the select2 things from the cloned element and init it again.

Fix should come soon: https://github.com/saltcorn/select2/pull/3

**Comment by yoson7** [comment]
Thanks.
Perhaps initialize the entire new subform?

**Comment by christianhugoch** [comment]
Yes, that was my first idea, too. But I think it's harder to load a fresh form from the server than just cloning it, and with this fix, we only need to change something in the plugin and not the host.

Perhaps we should keep this on the list as a refactoring.

**Comment by christianhugoch** [comment]
Can you try select2 0.5.2, perhaps with a clear cache reload

**Comment by yoson7** [comment]
Replace select2 version is by removing the plugin and reinstall it?
In case of container deployment, new container image required?

**Comment by christianhugoch** [comment]
Does the initial repeater element not work, or do you see the error on new repeater elements?

Have you tried to hit Ctrl + F5

**Comment by yoson7** [comment]
I did it from private browser window. 
Indeed, there is a problem with the element:

```
Refused to apply style from '<<URL>>' because its MIME type ('text/html') is not a supported stylesheet MIME type, and strict MIME checking is enabled.

```

**Comment by christianhugoch** [comment]
Hm, just rechecked it with the city, streets, common names structure

I'll try to make a public example on saltcorn.com tomorrow, so that we can see the same

**Comment by christianhugoch** [comment]
Can you take a look at this:

https://cities_streets_example.saltcorn.com/view/edit_city?id=1

do you have another structure?

**Comment by yoson7** [comment]
This is the general structure for that form (I encounter that problem in other forms as well):
![pic1](https://github.com/user-attachments/assets/dda3ab7e-5fd4-451a-b858-75babfd99076)

You don't need to read Hebrew, to see that there is duplication pointed by the arrows.
The top - on the right place, is act as regular select menu:
![image](https://github.com/user-attachments/assets/57845d46-29b8-4b8b-a380-83a092682083)

The second - displaced, is a broken select2:
![image](https://github.com/user-attachments/assets/b2367665-e313-45eb-9951-c8c8fa80ed57)

**Comment by christianhugoch** [comment]
One thing I see failing is when you have two select2 fields for the same table column (not sure if this makes sense).

Can you create a little toy example that I can access? Or if you want, you can send a backup as dropbox or google drive link to the mail of my profile.

**Comment by yoson7** [comment]
Tried to communicate to your email. so far no answer.

**Comment by yoson7** [comment]
I've upgraded to 2.5.3 and it's the same. 
There is regular select in the field position, and it's broken i the way the select window go beyond the page border.
And just below there is the selec2 as shown in the screenshots I uploaded above.

Please tell me how to make that txt file of the config.
